### PR TITLE
cut v2.35.0

### PR DESCRIPTION
**Pull request by Nexucis**
I don't know if we should mention https://github.com/prometheus/prometheus/pull/10608 in the Changelog. @csmarchbanks what do you think ? 

Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

**Comment by roidelapluie** [comment]
#10608  must be in the changelog, it is a bug introduced in 2.33.5.

**Comment by roidelapluie** [comment]
Can we remove * [ENHANCEMENT] Prometheus is built with Go 1.18. #10501

and instead add a note on top:

This Prometheus release is built with go1.18, which contains two noticeable changes related to TLS:
- [TLS 1.0 and 1.1 disabled by default client-side](https://go.dev/doc/go1.18#tls10). Prometheus users can override this with the `min_version` parameter of [tls_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tls_config).
- [Go 1.18 now rejects certificates signed with the SHA-1 hash function](https://go.dev/doc/go1.18#sha1). This doesn't apply to self-signed root certificates.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
 * [ENHANCEMENT] Prometheus is built with Go 1.18. #10501
 * [ENHANCEMENT] TSDB: more efficient sorting of postings read from WAL at startup. #10500
-* [ENHANCEMENT] Azure SD: Add metric to track Azure SD failures #10476
+* [ENHANCEMENT] Azure SD: Add metric to track Azure SD failures. #10476
 * [ENHANCEMENT] Azure SD: Add an optional `resource_group` configuration. #10365
 * [ENHANCEMENT] Kubernetes SD: Support `discovery.k8s.io/v1` `EndpointSlice` (previously only `discovery.k8s.io/v1beta1` `EndpointSlice` was supported). #9570
 * [ENHANCEMENT] Kubernetes SD: Allow attaching node metadata to discovered pods. #10080
 * [ENHANCEMENT] OAuth2: Support for using a proxy URL to fetch OAuth2 tokens. #10492
 * [ENHANCEMENT] Configuration: Add the ability to disable HTTP2. #10492
+* [ENHANCEMENT] Config: Support overriding minimum TLS version #10610
```
Comment (roidelapluie) [inline CHANGELOG.md:16]:
```suggestion
* [ENHANCEMENT] Config: Support overriding minimum TLS version. #10610
```

**Comment by Nexucis** [comment]
I think we can still keep `* [ENHANCEMENT] Prometheus is built with Go 1.18. #10501` and add the note you written.

**Review by roidelapluie — APPROVED** [review]
LGTM

**Comment by Nexucis** [comment]
thanks @roidelapluie

**Pull request merged**